src/aleph/storage.py

```diff
@@ -22,113 +24,207 @@
 LOGGER = logging.getLogger("STORAGE")
 
 
-async def json_async_loads(s: str):
+class ContentSource(str, Enum):
+    DB = "DB"
+    P2P = "P2P"
+    IPFS = "IPFS"
+    INLINE = "inline"
+
+
+@dataclass
+class StoredContent:
+    hash: str
+    source: Optional[ContentSource]
+
+
+@dataclass
+class RawContent(StoredContent):
+    value: bytes
+
+    def __len__(self):
+        return len(self.value)
+
+
+@dataclass
+class JsonContent(StoredContent):
+    value: Any
+    raw_content: bytes
```

Comment:
This name makes me think about `class RawContent` just above.
```suggestion
    raw_value: bytes
```

Reply:
Originally I used it to store the raw content, but then I realized it was more convenient to just store bytes since not all JSON contents originate from raw contents. Fixing.